### PR TITLE
HHH-15944 Joins on non-managed-type polymorphisms don't work

### PR DESCRIPTION
**Pull request by dreab8**
https://hibernate.atlassian.net/browse/HHH-15944

alternative solution to https://github.com/hibernate/hibernate-orm/pull/5999

### CODE REVIEW

File: hibernate-core/src/main/java/org/hibernate/query/hql/internal/QuerySplitter.java [hunk 1]
```diff
@@ -1037,6 +1381,27 @@ public SqmCreationOptions getCreationOptions() {
...
+			SqmFrom copy = sqmFromCopyMap.get( sqmFrom );
+			if ( copy != null ) {
+				return (X) copy;
+			}
+			else {
+				return (X) sqmFrom.accept( this );
+			}
+		}
+
+		private void visitJoins(SqmFrom<?, ?> join, SqmAttributeJoin copy) {
```
Comment (beikov) [inline hibernate-core/src/main/java/org/hibernate/query/hql/internal/QuerySplitter.java:1394]: Nice refactoring. I think you should be able to use that for all joins though i.e. `visitCrossJoin` etc.

Also, what about fetches and treats? Do we not support that in polymorphic queries? Might be worth adding an assert statement here then at least to make this clear.

**Review by beikov — COMMENTED** [review]
Looks like a nice first step!

File: hibernate-core/src/main/java/org/hibernate/query/hql/internal/QuerySplitter.java (hunk 1 above)
Comment (dreab8) [reply hibernate-core/src/main/java/org/hibernate/query/hql/internal/QuerySplitter.java]: done, fI added a test for fetches and they seems working, for treat I added an exception

**Review by beikov — APPROVED** [review]
LGTM

**Pull request merged**